lib/src/features/user/weight_repository.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:fit_and_healthy/shared/models/WeightEntry.dart';
+import 'package:fit_and_healthy/src/features/auth/app_user_model.dart';
+import 'package:fit_and_healthy/src/features/auth/auth_repository/firebase_auth_repository.dart';
+
+class WeightRepository {
+  final FirebaseFirestore _firestore = FirebaseFirestore.instance;
+  final FirebaseAuthRepository _authRepository;
+
+  WeightRepository(this._authRepository);
+
+  Future<void> addWeightEntry(WeightEntry entry) async {
+    final AppUser user = _authRepository.currentUser!;
+    if (user.firebaseUser == null)
+      throw Exception('No authenticated user found!');
+    final weightHistoryCollection = _firestore
+        .collection('users')
+        .doc(user.firebaseUser!.uid)
+        .collection('weightHistory');
+    await weightHistoryCollection.add(entry.toMap());
+  }
+
+  Future<List<WeightEntry>> getWeightHistory() async {
+    final AppUser user = _authRepository.currentUser!;
+    if (user.firebaseUser == null)
+      throw Exception('No authenticated user found!');
+    final weightHistoryCollection = _firestore
+        .collection('users')
+        .doc(user.firebaseUser!.uid)
+        .collection('weightHistory');
```

Comment:
We should do some filtering based on data or limit amout of results when fetching these i think. Unless you actually need all the history:

for ex:
```
const today = new Date();
const thirtyDaysAgo = new Date();
thirtyDaysAgo.setDate(today.getDate() - 30);

const weightEntriesRef = firestore.collection('users').doc(userId).collection('weightEntries');
const query = weightEntriesRef.where('date', '>=', thirtyDaysAgo);
```

Reply:
maybe even have a funcion that is:
```
function getWeightEntriesLastDays(int days) {
   [...]
   (Query from above)
}
```